firmware/quadruna/BMS/src/app/app_soc.c

```diff
@@ -79,48 +119,35 @@ float app_soc_getOcvFromSoc(float soc_percent)
 void app_soc_init(void)
 {
     stats.prev_current_A = 0.0f;
-    stats.soc_address    = DEFAULT_SOC_ADDR;
 
     // Asoc assumed corrupt until proven otherwise
     stats.is_corrupt = true;
     stats.charge_c   = -1;
 
     // A negative soc value will indicate to app_soc_Create that saved SOC value is corrupted
-    // float saved_soc_c = -1.0f;
-
-    // TODO: Update to SD Card logic
-    // if (app_eeprom_readSocAddress(&stats.soc_address) == EXIT_CODE_OK)
-    // {
-    //     if (app_eeprom_readMinSoc(stats.soc_address, &saved_soc_c) == EXIT_CODE_OK)
-    //     {
-    //         if (IS_IN_RANGE(0.0f, SERIES_ELEMENT_FULL_CHARGE_C * 1.25f, saved_soc_c))
-    //         {
-    //             stats.charge_c   = (double)saved_soc_c;
-    //             stats.is_corrupt = false;
-    //         }
-    //     }
-    // }
-    // else
-    // {
-    //     // If address corrupted, revert to default SOC address location
-    //     stats.soc_address = DEFAULT_SOC_ADDR;
-    // }
-
-    // Update the active address that SOC is stored at
-    // if (app_eeprom_updateSavedSocAddress(&stats.soc_address) != EEPROM_STATUS_OK)
-    // {
-    //     stats.soc_address = DEFAULT_SOC_ADDR;
-    // }
+    float saved_soc_c = -1.0f;
+
+#ifdef TARGET_EMBEDDED
+
+    if (app_soc_readSocFromSd(&saved_soc_c))
+    {
+        if (IS_IN_RANGE(0.0f, SERIES_ELEMENT_FULL_CHARGE_C * 1.25f, saved_soc_c))
+        {
+            stats.charge_c   = (double)saved_soc_c;
+            stats.is_corrupt = false;
+        }
+    }
+    else
+    {
+        // anything to do if fail to read it corrrectly from the SD?
```

Comment:
Everything that needs to be done is done already, set SOC corrupt and set SOC to -1 to signify it needs to be reset based on cell voltage